frb_rust/src/ffi/web.rs

```diff
@@ -0,0 +1,207 @@
+use super::DartCObject;
+use super::IntoDart;
+use super::MessagePort;
+pub use js_sys;
+pub use js_sys::Array as JsArray;
+use js_sys::{global, Array, ArrayBuffer};
+use std::iter::FromIterator;
+pub use wasm_bindgen;
+pub use wasm_bindgen::closure::Closure;
+pub use wasm_bindgen::prelude::*;
+pub use wasm_bindgen::JsCast;
+use web_sys::{DedicatedWorkerGlobalScope, Worker};
+
+pub trait IntoDartExceptPrimitive: IntoDart {}
+impl IntoDartExceptPrimitive for JsValue {}
+
+impl IntoDart for () {
+    #[inline]
+    fn into_dart(self) -> DartCObject {
+        JsValue::undefined()
+    }
+}
+macro_rules! delegate {
+    ($( $ty:ty )*) => {$(
+        impl IntoDart for $ty {
+            #[inline]
+            fn into_dart(self) -> DartCObject {
+                DartCObject::from(self)
+            }
+        }
+    )*};
+}
+macro_rules! delegate_buffer {
+    ($( $ty:ty => $buffer:ty )*) => {$(
+        impl IntoDart for $ty {
+            #[inline]
+            fn into_dart(self) -> DartCObject {
+                <$buffer>::from(self.as_slice()).into()
+            }
+        }
+    )*};
+}
+// Orphan rules disallow blanket implementations, so we have to manually delegate here.
+delegate! {
+    bool i8 u8 i16 u16 i32 u32 i64 u64 i128 u128 isize usize
+    f32 f64 &str String JsValue
+}
+delegate_buffer! {
+    Vec<i8> => js_sys::Int8Array
+    Vec<u8> => js_sys::Uint8Array
+    Vec<i16> => js_sys::Int16Array
+    Vec<u16> => js_sys::Uint16Array
+    Vec<i32> => js_sys::Int32Array
+    Vec<u32> => js_sys::Uint32Array
+    Vec<f32> => js_sys::Float32Array
+    Vec<f64> => js_sys::Float64Array
+    ZeroCopyBuffer<Vec<i8>> => js_sys::Int8Array
+    ZeroCopyBuffer<Vec<u8>> => js_sys::Uint8Array
+    ZeroCopyBuffer<Vec<i16>> => js_sys::Int16Array
+    ZeroCopyBuffer<Vec<u16>> => js_sys::Uint16Array
+    ZeroCopyBuffer<Vec<i32>> => js_sys::Int32Array
+    ZeroCopyBuffer<Vec<u32>> => js_sys::Uint32Array
+    ZeroCopyBuffer<Vec<f32>> => js_sys::Float32Array
+    ZeroCopyBuffer<Vec<f64>> => js_sys::Float64Array
+}
+
+impl<T: IntoDartExceptPrimitive> IntoDart for Vec<T> {
+    #[inline]
+    fn into_dart(self) -> DartCObject {
+        Array::from_iter(self.into_iter().map(IntoDart::into_dart)).into()
+    }
+}
+
+impl<T: IntoDart> IntoDart for Option<T> {
+    #[inline]
+    fn into_dart(self) -> DartCObject {
+        self.map(T::into_dart).unwrap_or_else(JsValue::null)
+    }
+}
+
+impl<T: IntoDart + Clone> IntoDart for &T {
+    #[inline]
+    fn into_dart(self) -> DartCObject {
+        self.clone().into_dart()
+    }
+}
+
+#[derive(Clone)]
+pub struct Isolate {
```

Comment:
Is that a real isolate in the sense of Dart language? If not, maybe (just like discussed above), we need to introduce a new umbrella name

Reply:
This is a misnomer: both the upstream `Isolate` and this file's does not actually hold an isolate, but only the port used to communicate with other isolates. How does `Channel` sound?

Reply:
I see. Channel looks better, as long as there is *no term* inside Dart/Flutter with that name so we do not have a name conflict and confusion.